### ISSUE TITLE
npm/node package security issues - update dependencies & older libraries 

### ISSUE THREAD

**Issue by joshc0044** [EpistasisLab/pennai#205]
several out of date node/npm packages are currently security issues of varying degrees 

update items in question

**Comment by joshc0044** [comment]
### small update from chat messages yesterday:

with a bit of trouble, npm audit appears to have worked in removing/mitigating many of the security vulnerabilities (needed to delete old `node_modules` folder, and run `npm install` and then `npm update` in order for `npm audit fix` to work correctly)

This was a quick test for the node packages in `lab/webapp/`

That is the extent of the automated fixing, there are
```found 83 low severity vulnerabilities in 58353 scanned packages
  72 vulnerabilities require semver-major dependency updates.
  11 vulnerabilities require manual review. See the full report for details.
```
after running the audit fix command, the remaining issues require manually updating the various packages, but the major issues were apparently resolved. I'm going through the project and rebuilding it, check the website and will run the test suites to see if everything is okay

I resolved some of the dependency vulnerabilities but it involved updating the babel & jest javascript libraries in the `lab\webapp` package.json which affect the web unit tests; the unit tests run with `docker-compose-unit-test.yml` ends up using a different version of babel, somehow it is using version `6.26.3` but I am able to run the tests successfully directly from a command prompt

I see in the unit test dockerfile that it copying the package.json from `lab\webapp` (which has the proper version of babel). I attempted to rebuild the unit test runner with `docker-compose -f .\docker-compose-unit-test.yml build` with and without the `no-cache` flag but it will error out when it goes to run the jest unit tests because of a wrong version of babel

Does anyone know where this version of babel is coming from? Locally, I had to remove my node_modules folder and delete the package-lock.json to get npm to install the packages correctly but I thought rebuilding without a cache would install the proper versions

Alternatively, I can avoid changing babel or jest versions (these result in low risk vulnerabilities) in place for now, using `npm audit fix` does address most of the severe/moderate issues successfully

**update for today**

I could not view the list of packages which were causing issues on the project repo until yesterday afternoon, I was trying to resolve all the issues by updating the packages in question reported back from `npm audit` (permissions for the project's github repo)

Now, I can view the list of packages which need updating which should simplify things a bit as I should not need to touch the versions for testing framework (jest) and the transpiler (babel) - these were causing some issues with the docker unit test runner. I will address the packages outlined in the security alert

**Comment by joshc0044** [comment]
**5/28/2019**

Last week pushed changes which resolved security/vulnerability issues for github - closing out this ticket for now

**Comment by weixuanfu** [comment]
6/24 Most of security/vulnerability are fixed in PR #215, #212 and commit 1935b54777854a29ffbcb9e3522b32119796c942. So I closed this issue.
